java/cuvs-java/src/main/java22/com/nvidia/cuvs/spi/JDKProvider.java

```diff
@@ -129,19 +149,32 @@ private void internalAddVector(Object vector) {
         MemorySegment.copy(
             vector,
             0,
-            dataset.memorySegment(),
-            dataset.valueLayout(),
-            ((current++) * dimensions * dataset.valueLayout().byteSize()),
-            dimensions);
+            matrix.memorySegment(),
+            matrix.valueLayout(),
+            ((current++) * columns * matrix.valueLayout().byteSize()),
+            columns);
       }
 
       @Override
       public CuVSMatrix build() {
-        return dataset;
+        return matrix;
       }
     };
   }
 
+  @Override
+  public CuVSMatrix.Builder newDeviceMatrixBuilder(
+      CuVSResources resources, int size, int columns, CuVSMatrix.DataType dataType, int copyType)
+      throws UnsupportedOperationException {
+
+    var builderCopyType = copyType & 0xF0;
```

Comment:
If you wonder what this is, see https://github.com/rapidsai/cuvs/pull/1232/files#r2263476356

TL;DR: different implementations to benchmark, only one will make into the final version, and the flags/switch will be removed too.